src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -500,9 +500,9 @@ internal void ExecutePackageDownload(string name, PackageVersion package, string
             {
                 System.Diagnostics.Debug.Assert(package.full_dependency_ids.Count == package.full_dependency_versions.Count);
                 // get all of the dependency version headers
-                var dependencyVersionHeaders = package.full_dependency_ids.Select((dep, i) =>
+                var dependencyVersionHeaders = package.full_dependency_ids.Select(x=>x).Reverse().Select((dep, i) =>
                 {
-                    var depVersion = package.full_dependency_versions[i];
+                    var depVersion = package.full_dependency_versions.Select(x => x).Reverse().ElementAt(i);
```

Comment:
yes, you're right, we can do the reversal once above and access it within the select statement will make the change.

Reply:
@mjkkirschner is the order of the dependencies documented on PM now ? and maybe covered by tests ?

Reply:
also please add a comment here ...to explain why we are Reversing the array

Reply:
I agree with @aparajit-pratap . Maybe we should reverse both arrays first...then parse them...

Reply:
yes, @pinzart90 in this PR: https://git.autodesk.com/Dynamo/PackageManager/pull/334 I moved to using a `map` because it guarantees the order of its keys, and added multiple tests to assert order of dependencies is returned correctly.